contracts/FFPaymentSplit.sol

```diff
@@ -94,12 +139,42 @@ contract FFPaymentSplit is PaymentSplitter {
         // Check is required here to prevent reentrancy from the Kudos contract
         if (msg.sender != address(_ffKudosInterface)) {
             require(address(_ffKudosInterface) != address(0), "FFPaymentSplitter: NFT contract is not set");
-            _donationIds.increment();
+
+            uint evaluatorId = uint(uint8(msg.data[0]));
```

Comment:
@KiChjang Can clarify what is happening here, or why this is necessary?

Reply:
This is the tough part that I was mentioning in Gitcoin Chat -- basically, the `msg.data` is a byte array, and here I'm extracting the first position out with the `[0]` syntax, which would give me a `byte` type, and right after that I'm typecasting it to a `uint8`, and then finally a `uint`.

This is all technobabble for saying "I just wanna extract a uint256 from `msg.data` and this is how I've done it".

Reply:
Thanks, I didn't understand why it was necessary to first cast to `uint8` and then to `uint`. Bring on the technobabble!